.github/workflows/iroha2-dev-pr.yml

```diff
@@ -44,24 +48,49 @@ jobs:
     container:
       image: 7272721/i2-ci:latest
     steps:
-      # TODO Remove these steps #2165
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Find hash
+        id: hash
+        uses: theowenyoung/folder-hash@v2.0.1
+        with:
+          path: |
+            .
```

Comment:
Should we remove all steps related to codecov reports files hash identification & comparison and artifacts uploading & downloading as well?

Reply:
This approach is based on creating the unique codecov coverage report according to the repository state. This should allows to reuse codevov coverage report for the same repository (PR) state. So, confirm again if it's not needed.

Reply:
Assuming the same repository state means the same folder hash, that means the `on.pull_request.paths` is not triggered, which means the entire workflow is skipped. If so, when will the report be reused?

Reply:
It was useful for cross-workflows reuse until [this PR was merged](https://github.com/hyperledger/iroha/commit/219355e552eb3cb7ac9614994cf71f21ab6c9ebb). I'm not sure if this is still relevant.

Reply:
Now that in fdac826 the push workflow reuse the coverage report, something like the repository hash will be required for identifying the report (I don't think `github.sha` works for this case)

Reply:
"I don't think github.sha works for this case" : we don't use `github.sha`, but we try to calculate files & folders hash.

Reply:
Yeah, I meant I agreed with the use of `theowenyoung/folder-hash`.

After researching, however, I found a way to avoid the hash calculation -- upload the report with `github.sha` on the PR workflow and download it with `github.event.pull_request.head.sha` on the ~~push~~ [merge workflow](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges):

`iroha2-dev-pr.yml`

```yml
jobs:
  # omit
  tests_with_coverage:
    # omit
    steps:
      # omit
      - name: Upload codecov report artifact
        uses: actions/upload-artifact@v3
        with:
          name: lcov-${{ github.sha }}.info
          path: lcov.info

```

`iroha2-dev.yml`

```yml
on:
  pull_request:
    branches: [iroha2-dev]
    types:
      - closed
# omit
jobs:
  # omit
  upload_coverage:
    if: github.event.pull_request.merged
    # omit
    steps:
      # omit
      - name: Download codecov report artifact
        uses: dawidd6/action-download-artifact@v2.21.1
        with:
          workflow: iroha2-dev-pr.yml
          name: lcov-${{ github.event.pull_request.head.sha }}.info

```

Reply:
@BAStos525 I like the approach, the less we depend on external actions, the better.

Reply:
I found out that `dawidd6/action-download-artifact@v2.21.1` Actioon fails on  `name: lcov-${{ github.event.pull_request.head.sha }}.info`
It works if just ti set the parameter to download the last  available artifact from previous workflow. Fortunately, it matches with created artifact on PR workflow with `name: lcov-${{ github.sha }}.info`
But probably expected `coverage diff` still doesn't work.

Reply:
I hope https://github.com/hyperledger/iroha/pull/2433#discussion_r922151913 resolves this